src/crawlee/memory_storage_client/_key_value_store_client.py

```diff
@@ -287,6 +287,12 @@ async def delete_record(self, key: str) -> None:
                 if self._memory_storage_client.persist_storage:
                     await existing_store_by_id.delete_persisted_record(record)
 
+    @override
+    async def get_public_url(self, key: str) -> str:
+        store_name = self.name or self.id
+        storage_dir = self._memory_storage_client.storage_dir
+        return f'file://{storage_dir}/key_value_stores/{store_name}/{key}.txt'
```

Comment:
This is... surprising. Since the tests appear to pass, I guess that's because of some weird implementation detail, but I don't think that `set_value` should automatically add the `.txt` suffix.

Reply:
Ya not sure of the reason, but the file stored in my system also gets the .txt extension. I could not understand the reason of test failure when I wrote it without .txt, and then navigated to the directory to find it

<img width="770" alt="image" src="https://github.com/user-attachments/assets/aa325264-333a-463a-a115-6388e457f9c3">

Reply:
@vdusek can you help out here? It looks like the key-value store adds a txt suffix to files with `content_type = None`

Reply:
@janbuchar @vdusek  I took a quick look at it... Here the code determined the content type to be `text/plain`
https://github.com/apify/crawlee-python/blob/fb96524687c54f53f6f65b99f9574c1d777dff7a/src/crawlee/memory_storage_client/_key_value_store_client.py#L240-L246 
Afterwards at this point the extension for `text/plain` is determined to be `.txt` https://github.com/apify/crawlee-python/blob/fb96524687c54f53f6f65b99f9574c1d777dff7a/src/crawlee/_utils/file.py#L76
which is called from https://github.com/apify/crawlee-python/blob/fb96524687c54f53f6f65b99f9574c1d777dff7a/src/crawlee/memory_storage_client/_key_value_store_client.py#L260

And this is then eventually saved with `.txt` extension.

Reply:
I don't see a good reason for keyvalue store to add any extension to the objects, but I'd like to hear your opinion @vdusek

Reply:
@akshay11298 Thanks for your investigations. I agree with @janbuchar on this - the KVS should not try to determine the extension and append it.

Reply:
@vdusek Updated the code to not determine content_type if None is passed

Reply:
I investigated the issue some more and decided that we probably shouldn't change the internal workings of the memory storage too much. Instead we can just do this:

```py
    @override
    async def get_public_url(self, key: str) -> str:
        store_name = self.name or self.id
        storage_dir = self._memory_storage_client.storage_dir
        record = await self._get_record_internal(key)

        return f'file://{storage_dir}/key_value_stores/{store_name}/{record.filename if record else key}'
```

It will use an internal methods that does a lookup based on the original key, and then you can return the filename used by the key-value store.

The extension handling code still seems unnecessary, but we'll leave it for now.

Reply:
@akshay11298 would you mind reverting the latest commit and doing what I propose instead?